src/jit/hwintrinsicxarch.cpp

```diff
@@ -754,6 +760,70 @@ GenTree* Compiler::impSSE2Intrinsic(NamedIntrinsic        intrinsic,
             }
             break;
 
+        case NI_SSE2_ConvertScalarToVector128Double:
+        {
+            assert(sig->numArgs == 2);
+            assert(getBaseTypeOfSIMDType(sig->retTypeSigClass) == TYP_DOUBLE);
+
+            argList = info.compCompHnd->getArgNext(sig->args);
+            CorInfoType corType =
+                strip(info.compCompHnd->getArgType(sig, argList, &argClass)); // type of the second argument
+
+            baseType = JITtype2varType(corType);
+
+#ifdef _TARGET_X86_
+            if (varTypeIsLong(JITtype2varType(corType)))
+            {
+                return impUnsupportedHWIntrinsic(CORINFO_HELP_THROW_PLATFORM_NOT_SUPPORTED, method, sig, mustExpand);
+            }
+#endif // _TARGET_X86_
+
+            if (baseType == TYP_STRUCT)
+            {
+                baseType = getBaseTypeOfSIMDType(argClass);
+                op2      = impSIMDPopStack(TYP_SIMD16);
+            }
+            else
+            {
+                op2 = impPopStack().val;
+            }
+
+            op1     = impSIMDPopStack(TYP_SIMD16);
+            retNode = gtNewSimdHWIntrinsicNode(TYP_SIMD16, op1, op2, intrinsic, baseType, simdSize);
+
+            break;
+        }
+
+        case NI_SSE2_ConvertScalarToVector128Single:
+        {
+            assert(sig->numArgs == 2);
+            assert(getBaseTypeOfSIMDType(sig->retTypeSigClass) == TYP_FLOAT);
+
+            argList = info.compCompHnd->getArgNext(sig->args);
+            CorInfoType corType =
+                strip(info.compCompHnd->getArgType(sig, argList, &argClass)); // type of the second argument
+
+            baseType = getBaseTypeOfSIMDType(argClass);
```

Comment:
The baseType is `float` (retType's base type) on `ConvertScalarToVector128Single`.

Reply:
I need arg types to use correct encoding later (r64 versus r32) during instruction emitting.

Reply:
Are you talking about `ConvertScalarToVector128Double`?

Reply:
Once I will work on refactoring I will have overloaded - where I need to use second argType as base type
`ConvertScalarToVector128Single(Vector128<float> upper, int value)`
`ConvertScalarToVector128Single(Vector128<float> upper, long value)`

If you prefer to simplify it for this PR and not defer changes to refactoring I will do it.

Reply:
I see. However, I do not think we should combine `Sse.ConvertScalarToVector128Single` and `Sse2.ConvertScalarToVector128Single` because they have different signatures and codgen consideration.

Furthermore, combining HW intrinsic code cross-ISA should based-on the current category/flag mechanism rather than "intrinsic names". Meanwhile, a little bit duplicated code is fine to me, it may be not worthwhile to complicate the framework just for a few special cases. We can implement these "special cases" individually, then add categories/flags to combine them if there are some obvious patterns.

Reply:
OK than I will simplify it and leave any changes during refactoring to the result of discussion of its scope.